packages/demo-nextjs/src/components/Menu/components/SaveButton/SaveButton.tsx

```diff
@@ -0,0 +1,37 @@
+import { APIActionState, APIActionStateType } from "@/components/Menu/utils";
+import { KTIcon } from "@/components/Metronic/helpers";
+import clsx from "clsx";
+import React, { MouseEventHandler } from "react";
+
+export function SaveButton({
+  handleSaveMenu,
+  saveState,
+}: {
+  handleSaveMenu: MouseEventHandler;
+  saveState: APIActionStateType;
+}) {
+  const className = clsx(
+    "btn",
+    "mt-8",
+    saveState === APIActionState.idle && "btn-primary",
+    saveState === APIActionState.success && "btn-success",
+    saveState === APIActionState.error && "btn-danger",
+    saveState === APIActionState.loading && "btn-secondary",
```

Comment:
the buttons needs to have separated states don't need to be the same as reducers state